### DISCUSSION TITLE
Connect wrong DB connection to query data when executing queue

### DISCUSSION THREAD

**Discussion by Twinkle0613** [laravel/framework#45659] [Q&A]
Laravel Version: 8.83.26
PHP Version: 7.4.33
Database Driver & Version: MySQL 5.7

**Description:**
 In my Laravel application, I setup two DB ('base' and 'log') for storing operation data and log data as shown below.
Sometimes, during the execution queue, the system connects to the wrong database to execute the query. The exception generated is shown below. It won't happen in every queues. 

By right, this query need to connect 'base'
`
"message": "SQLSTATE[42S02]: Base table or view not found: 1146 Table 'base.user_logs' doesn't exist (SQL: select * from 'user_logs' where 'user_id' = 1 and 'date' between 2023-01-16 02:11:03 and 2023-01-16 02:26:06 limit 100)"
`

config/database.php

        'base' => [
            'read' => [
                'host' => [
                    env('DB_HOST_READ', '127.0.0.1'),
                ],
                'database' => env('DB_DATABASE_READ', 'forge'),
                'username' => env('DB_USERNAME_READ', 'forge'),
                'password' => env('DB_PASSWORD_READ', ''),
            ],
            'write' => [
                'host' => [
                    env('DB_HOST_WRITE', '127.0.0.1'),
                ],
                'database' => env('DB_DATABASE_WRITE', 'forge'),
                'username' => env('DB_USERNAME_WRITE', 'forge'),
                'password' => env('DB_PASSWORD_WRITE', ''),
            ],
            'driver' => 'mysql',
            'url' => env('DATABASE_URL'),
            'host' => env('DB_HOST', '127.0.0.1'),
            'port' => env('DB_PORT', '3306'),
            'database' => env('DB_DATABASE', 'forge'),
            'username' => env('DB_USERNAME', 'forge'),
            'password' => env('DB_PASSWORD', ''),
            'unix_socket' => env('DB_SOCKET', ''),
            'charset' => 'utf8mb4',
            'collation' => 'utf8mb4_unicode_ci',
            'prefix' => '',
            'prefix_indexes' => true,
            'strict' => true,
            'engine' => null,
        ],

        'log' => [
            'read' => [
                'host' => [
                    env('DB_LOG_HOST_READ', '127.0.0.1'),
                ],
                'database' => env('DB_LOG_DATABASE_READ', 'forge'),
                'username' => env('DB_LOG_USERNAME_READ', 'forge'),
                'password' => env('DB_LOG_PASSWORD_READ', ''),
            ],
            'write' => [
                'host' => [
                    env('DB_HOST_WRITE', '127.0.0.1'),
                ],
                'database' => env('DB_LOG_DATABASE_WRITE', 'forge'),
                'username' => env('DB_LOG_USERNAME_WRITE', 'forge'),
                'password' => env('DB_LOG_PASSWORD_WRITE', ''),
            ],
            'driver' => 'mysql',
            'url' => env('DATABASE_LOG_URL'),
            'host' => env('DB_LOG_HOST', '127.0.0.1'),
            'port' => env('DB_LOG_PORT', '3306'),
            'database' => env('DB_LOG_DATABASE', 'forge'),
            'username' => env('DB_LOG_USERNAME', 'forge'),
            'password' => env('DB_LOG_PASSWORD', ''),
            'unix_socket' => env('DB_LOG_SOCKET', ''),
            'charset' => 'utf8mb4',
            'collation' => 'utf8mb4_unicode_ci',
            'prefix' => '',
            'prefix_indexes' => true,
            'strict' => true,
            'engine' => null,
        ]

For addition informations:
1) Both "base" connection and "log" connection have been setup 'write' and 'read'. 
2) Queued jobs are stored into Redis.

Anyone know how this error would occur with this system?

Thanks

**Comment by henzeb** [comment]
Maybe you forgot to specify the connection in an conditionally executed statement?
